### PR TITLE
Enhancement: Implementation of getAbsentLesson ✅

### PR DESCRIPTION
**Pull request by Jauki**
# Absent Lesson
This is an Implementation to get the absent Lesson of a Student. The purpose is to have an array of absent Lessons which you can view. 

## Route
`/WebUntis/api/classreg/absences/students`
I did a network analysis on the official WebUntis site and watched where the route was mapped. Then implemented a simple function where I have 4 parameters: `validateSession, rangeStart, rangeEnd, excuseStatusId`

`rangeStart` is usually the start of the School year for Example -> 13.09.2021 (Austria, Styria)
`rangeEnd` is usually the current Date

## Additional
I needed a function like this, but didn't want to build the whole thing from scratch. ✌️

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -65,3 +65,6 @@ typings/
 test2.js
 
 .vscode
+
+package-lock.json
+yarn.lock
```
Comment (TheNoim) [inline .gitignore:70]: [Do not ignore yarn.lock](https://github.com/yarnpkg/yarn/issues/1583#issuecomment-257544217)

File: index.js [hunk 2]
```diff
@@ -744,6 +744,33 @@ class WebUntis {
...
+     * Returns all the Lessons where you were absent including the excused one!
+     * @param {Date} rangeStart
+     * @param {Date} rangeEnd
+     * @param {Integer} execuseStatusId
+     * @param {boolean} [validateSession=true]
+     * @returns {Promise.<Array>}
+     */
+    async getAbsentLesson(validateSession = true, rangeStart, rangeEnd, excuseStatusId = -1) {
+        if (validateSession && !(await this.validateSession())) throw new Error('Current Session is not valid');
+        const response = await this.axios({
```
Comment (TheNoim) [inline index.js:758]: You should add `this._checkAnonymous();`

**Review by TheNoim — CHANGES_REQUESTED** [review]

**Comment by Jauki** [comment]
Fixed those minor changes!

File: index.js [hunk 3]
```diff
@@ -744,6 +744,81 @@ class WebUntis {
...
+
+    /**
+     * Returns all the Lessons where you were absent including the excused one!
+     * @param {Date} rangeStart
+     * @param {Date} rangeEnd
+     * @param {Integer} execuseStatusId
+     * @param {boolean} [validateSession=true]
+     * @returns {Promise.<Array>}
+     */
+    async getAbsentLesson(validateSession = true, rangeStart, rangeEnd, excuseStatusId = -1) {
```
Comment (TheNoim) [inline index.js:756]: Sorry for only noticing now, but you forgot to add type definitions (index.d.ts) for the new function.

**Review by TheNoim — CHANGES_REQUESTED** [review]

File: index.js [hunk 4]
```diff
@@ -770,6 +770,7 @@ class WebUntis {
             },
         });
         if (response.data.data == null) throw new Error('Server returned no data!');
+        console.log(response.data.data);
```
Comment (TheNoim) [inline index.js:773]: I guess, this shouldn't be here.

File: index.js [hunk 5]
```diff
@@ -744,6 +744,82 @@ class WebUntis {
...
+     * @param {Date} rangeEnd
+     * @param {Integer} execuseStatusId
+     * @param {boolean} lateness
+     * @param {boolean} absences
+     * @param {boolean} [validateSession=true]
+     * @param {boolean} execuseGroup
+     * @returns {String} URL
+     */
+    async getPdfOfAbsentLesson(
+        validateSession = true,
```
Comment (TheNoim) [inline index.js:789]: > I suggest adding parameters with default values at the end of all arguments. 

.

**Review by TheNoim — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (TheNoim) [reply index.js]: Don't forget to update the type definitions ^^

Comment (Jauki) [reply index.js]: Where should i add the type definitions?

Comment (TheNoim) [reply index.js]: You forgot to move the parameter https://github.com/SchoolUtils/WebUntis/pull/76/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R446

Comment (Jauki) [reply index.js]: I see! thx

**Pull request merged**